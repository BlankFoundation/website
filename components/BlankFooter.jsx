import {
  TWFooter,
  TWHorizontal,
  NewWindowLink,
  NextLink,
} from '.'

const BlankFooter = ({ social }) => {
  return (
    <TWFooter className='sm:px-24'>
      <TWHorizontal className='mx-auto sm:mx-0 space-x-8'>
        {social.map((s, index) => (
          <NewWindowLink
            key={`footer-social-${index}`} 
            href={s.href}
          >
            {s.name}
          </NewWindowLink> 
        ))}
      </TWHorizontal>
      <NextLink 
        href='/mint' 
        passHref
      >
        <a className='hidden sm:inline-block text-white'>
          Mint
        </a>
      </NextLink>
    </TWFooter>
  );
}

BlankFooter.defaultProps = {
  index: "index",
  social: [{"href":"https://discord.gg/HZM2jcHdEU","name":"Discord"},{"href":"https://twitter.com/blankfdn","name":"Twitter"},{"href":"https://medium.com/@blankfdn","name":"Medium"}]
}

export default BlankFooter;