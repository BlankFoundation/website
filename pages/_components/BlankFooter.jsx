import {
  TWFooter,
  TWHorizontal,
  NewWindowLink
} from '.'

const BlankFooter = ({ index, social }) => {
  return (
    <TWFooter>
      <TWHorizontal className='space-x-4'>
        {social.map((s, index) => (
          <NewWindowLink
            key={`footer-social-${index}`} 
            href={s.href}
          >
            {s.name}
          </NewWindowLink> 
        ))}
      </TWHorizontal>
    </TWFooter>
  );
}

BlankFooter.defaultProps = {
  index: "index",
  social: [{"href":"https://discord.gg/HZM2jcHdEU","name":"Discord"},{"href":"https://twitter.com/blankfdn","name":"Twitter"},{"href":"https://medium.com/@blankfdn","name":"Medium"}]
}

export default BlankFooter;