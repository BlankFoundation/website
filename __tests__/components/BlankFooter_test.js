import React from 'react';
import TestRenderer from 'react-test-renderer';
import BlankFooter from '../../components/BlankFooter';

describe("BlankFooter", () => {
  it('renders correctly', () => {
    const rendered = TestRenderer.create(<BlankFooter />).toJSON();
    expect(rendered).toMatchSnapshot()
  });
});