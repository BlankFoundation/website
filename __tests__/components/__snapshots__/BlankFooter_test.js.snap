// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankFooter renders correctly 1`] = `
<div
  className="sm:px-24 p-3 bg-gray-900 text-white text-xs leading-10"
>
  <div
    className="sm:px-24 justify-between flex flex-row"
  >
    <div
      className="mx-auto sm:mx-0 space-x-8 flex flex-row"
    >
      <a
        href="https://discord.gg/HZM2jcHdEU"
        rel="noopener noreferrer"
        target="_blank"
      >
        Discord
      </a>
      <a
        href="https://twitter.com/blankfdn"
        rel="noopener noreferrer"
        target="_blank"
      >
        Twitter
      </a>
      <a
        href="https://medium.com/@blankfdn"
        rel="noopener noreferrer"
        target="_blank"
      >
        Medium
      </a>
      <a
        href="https://blankfoundation.notion.site/Blank-641ad836df9c4918b1bd09ce196a6dce"
        rel="noopener noreferrer"
        target="_blank"
      >
        Notion
      </a>
    </div>
    <a
      className="hidden sm:inline-block text-white"
      href="/mint"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Mint
    </a>
  </div>
</div>
`;
